src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TeleporterListener.java

```diff
@@ -34,31 +34,32 @@ public void onPressurePlateEnter(PlayerInteractEvent e) {
             return;
         }
 
-        String id = BlockStorage.checkID(e.getClickedBlock());
+        SlimefunItem item = BlockStorage.check(e.getClickedBlock());
 
-        if (id == null) {
+        // Fixes #2966
+        if (item == null || item.isDisabledIn(e.getPlayer().getWorld())) {
             return;
         }
 
-        if (isTeleporterPad(id, e.getClickedBlock(), e.getPlayer().getUniqueId())) {
+        if (isTeleporterPad(item, e.getClickedBlock(), e.getPlayer().getUniqueId())) {
             SlimefunItem teleporter = BlockStorage.check(e.getClickedBlock().getRelative(BlockFace.DOWN));
 
             if (teleporter instanceof Teleporter && checkForPylons(e.getClickedBlock().getRelative(BlockFace.DOWN))) {
                 Block block = e.getClickedBlock().getRelative(BlockFace.DOWN);
                 UUID owner = UUID.fromString(BlockStorage.getLocationInfo(block.getLocation(), "owner"));
                 SlimefunPlugin.getGPSNetwork().getTeleportationManager().openTeleporterGUI(e.getPlayer(), owner, block, SlimefunPlugin.getGPSNetwork().getNetworkComplexity(owner));
             }
-        } else if (id.equals(SlimefunItems.ELEVATOR_PLATE.getItemId())) {
+        } else if (item instanceof ElevatorPlate) {
             ElevatorPlate elevator = ((ElevatorPlate) SlimefunItems.ELEVATOR_PLATE.getItem());
             elevator.openInterface(e.getPlayer(), e.getClickedBlock());
         }
     }
 
     @ParametersAreNonnullByDefault
-    private boolean isTeleporterPad(String id, Block b, UUID uuid) {
-        if (id.equals(SlimefunItems.GPS_ACTIVATION_DEVICE_SHARED.getItemId())) {
+    private boolean isTeleporterPad(SlimefunItem item, Block b, UUID uuid) {
+        if (item.getId().equals(SlimefunItems.GPS_ACTIVATION_DEVICE_SHARED.getItemId())) {
             return true;
-        } else if (id.equals(SlimefunItems.GPS_ACTIVATION_DEVICE_PERSONAL.getItemId())) {
+        } else if (item.getId().equals(SlimefunItems.GPS_ACTIVATION_DEVICE_PERSONAL.getItemId())) {
```

Comment:
This hardcoded id check is really not optimal.

Is there a Teleporter Pad class? If not we should probably add that and a `hasAccess(Player, Block)` method then to allow modifications.

Reply:
Refactored in https://github.com/Slimefun/Slimefun4/commit/a3a9ee4b909a98696e96d1bd4c5551204986ec18